examples/vision/keypoint_detection.py

```diff
@@ -0,0 +1,440 @@
+in our face include nose tips, eyebrows, eye corners, and so on. These parts help to
+represent the underlying object in a feature-rich manner. Keypoint detection has
+applications such as pose estimation, face detection, etc.
+
+In this example, we will build a keypoint detector using the [StanfordExtra dataset](https://github.com/benjiebob/StanfordExtra)
+with transfer learning. This example requires TensorFlow 2.4 or higher, as well as [`imgaug`](https://imgaug.readthedocs.io/),
+which can be installed using the following command:
+"""
+
+"""shell
+pip install -q -U imgaug
+"""
+
+"""
+## Data collection
+"""
+
+"""
+The StanfordExtra dataset contains 12000 images to dogs with their keypoints and
+segmentation maps labeled. It is developed from the [Stanford dogs dataset](http://vision.stanford.edu/aditya86/ImageNetDogs/).
+It can be downloaded with the command below:
+"""
+
+"""shell
+wget -q http://vision.stanford.edu/aditya86/ImageNetDogs/images.tar
+"""
+
+"""
+Annotations are provided as a single JSON file in the StanfordExtra dataset and one needs
+to fill [this form](https://forms.gle/sRtbicgxsWvRtRmUA) to get access to it. Since the
+authors explicitly instruct about not sharing the JSON file, this example respects that
+and does not provide it. The JSON file will download as `stanfordextra_v12.zip`. 
+
+After the files are downloaded we can extract the archives. 
+"""
+
+"""shell
+tar xf images.tar
+unzip -qq stanfordextra_v12.zip
+rm -rf images.tar stanfordextra_v12.zip
+"""
+
+"""
+## Imports
+"""
+
+from tensorflow.keras import layers
+from tensorflow import keras
+import tensorflow as tf
+
+from imgaug.augmentables.kps import KeypointsOnImage
+from imgaug.augmentables.kps import Keypoint
+import imgaug.augmenters as iaa
+
+from PIL import Image
+from sklearn.model_selection import train_test_split
+from matplotlib import pyplot as plt
+import pandas as pd
+import numpy as np
+import json, os
+
+"""
+## Define hyperparameters
+"""
+
+IMG_SIZE = 224
+BATCH_SIZE = 64
+EPOCHS = 5
+NB_KEYPOINTS = 24 * 2  # 24 pairs each having x and y coordinates
+
+"""
+## Load data
+"""
+
+"""
+The authors also provide a metadata file that specifies additional information about the
+keypoints like color information, animal pose name, etc. We will load that in a `pandas`
+dataframe to extract information for visualization purposes. 
+"""
+
+IMG_DIR = "Images"
+JSON = "StanfordExtra_V12/StanfordExtra_v12.json"
+KEYPOINT_DEF = (
+    "https://github.com/benjiebob/StanfordExtra/raw/master/keypoint_definitions.csv"
+)
+
+# Load the ground-truth annotations.
+with open(JSON) as infile:
+    json_data = json.load(infile)
+
+# Set up a dictionary, mapping all the ground-truth information
+# with respect to the path of the image.
+json_dict = {i["img_path"]: i for i in json_data}
+
+"""
+A single entry of `json_dict` looks like the following:
+
+```
+'n02085782-Japanese_spaniel/n02085782_2886.jpg': {'img_bbox': [205,
```

Comment:
Please pretty-print the JSON dict

Reply:
You can find it [here](https://colab.research.google.com/drive/1VY8CxDRW7PFw-ivseSx7bgoWpTnrQWOk?usp=sharing).